ricecooker/utils/youtube_cache.py

```diff
@@ -0,0 +1,167 @@
+import json
+import os
+import re
+import youtube_dl
+
+from pressurecooker.youtube import YouTubeResource
+from ricecooker.config import LOGGER
+
+
+# CONSTANTS for YouTube cache
+################################################################################
+CHEFDATA_DIR = 'chefdata'
+DEFAULT_YOUTUBE_CACHE_DIR = os.path.join(CHEFDATA_DIR, 'youtubecache')
+
+
+# CONSTANTS for YouTube resources
+################################################################################
+YOUTUBE_VIDEO_REGEX = re.compile(
+    r'(https?://)?(www\.)?(youtube|youtu|youtube-nocookie)\.(com|be)/(watch\?v=|embed/|v/|.+\?v=)?(?P<youtube_id>[A-Za-z0-9\-=_]{11})'
+)
+YOUTUBE_PLAYLIST_URL_FORMAT = "https://www.youtube.com/playlist?list={0}"
+YOUTUBE_VIDEO_URL_FORMAT = "https://www.youtube.com/watch?v={0}"
+
+
+class YouTubeVideoCache(object):
```

Comment:
Let's just call this `YouTubeVideo` and note in the documentation that it caches information retrieved from the YT API. This is a wrapper around the YT API that has caching options, rather than a class just for caching. Let's also just have the filename be `youtube.py`for the same reason.